Issue 8

adashe opened the pull request:
Issue Reference #8

I documented/described the setup of the virtual environment, adding my documentation to directories 00 and 01 as 00.ad.documentation.md and 01.ad.documentation.md. Also added venv and requirements.txt. 

I used `pip3 install flask` in place of `sudo apt install python3-flask` when working from 00.prerequisites/README.md.

asokolsky commented on 00.prerequisites/00.ad.documentation.md:
```diff
@@ -0,0 +1,105 @@
+Checking which version of Python is installed:
```
I do not see benefits in adding another file to this directory.  If you would like to improve something - please state the problem and add information under releavant file name or into README

asokolsky commented on 00.prerequisites/00.ad.documentation.md:
```diff
@@ -0,0 +1,105 @@
+Installing collected packages: MarkupSafe, Werkzeug, urllib3, Jinja2, itsdangerous, idna, click, charset-normalizer, certifi, blinker, requests, flask
+Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 Werkzeug-2.3.6 blinker-1.6.2 certifi-2023.7.22 charset-normalizer-3.2.0 click-8.1.6 flask-2.3.2 idna-3.4 itsdangerous-2.1.2 requests-2.31.0 urllib3-2.0.4`
+
+Saved requirements.txt:
```
Note that your `requirements.txt` now has all the component versions fixed. 
You may want tot hink about benefits and problems this may cause....

asokolsky commented on 01.flask/01.ad.documentation.md:
```diff
@@ -0,0 +1,34 @@
+cd'ed into directory and ran `./helloworld -h`: 
```
As mentioned before I do not see a point in this file name.  If you are trying to document a particular task, please name the file accortdingly.

asokolsky commented on requirements.txt:
```diff
@@ -0,0 +1,12 @@
+blinker==1.6.2
```
https://stackoverflow.com/questions/55052434/does-python-requirements-file-have-to-specify-version

asokolsky reviewed (changes requested):
* Revise file names - make these relevant to a specific tasks
* remove package versions from requirements.txt

asokolsky commented:
FYI: use of make for venv management: https://asokolsky.github.io/apps/make/

adashe commented:
Updates made to Issue #8.

- Updated requirements.txt to remove versions
- Added comments about pip3 to install Flask instead of apt
- Reordered so virtual environment instructions are before Flask installation

asokolsky commented on 00.prerequisites/README.md:
````diff
@@ -57,9 +57,30 @@ If you have a new Linux install, just do:
+```
+alex@latitude7490:~/Projects/RESTing-with-Flask/02.rest/farm/ > python3 -m venv venv
+alex@latitude7490:~/Projects/RESTing-with-Flask/02.rest/farm/ > source ./venv/bin/activate
+(venv) alex@latitude7490:~/Projects/RESTing-with-Flask/02.rest/farm/ > pip3 install flask requests
````
Now that the `requirements.txt` is included:
```
pip install -r requirements.txt
```
More on this: https://note.nkmk.me/en/python-pip-install-requirements/

adashe replied:
I added comments to use `pip3 install -r requirements.txt` when there is requirements.txt available. 

I tried both pip3 and pip and they produced the same result. Google says Python3 uses pip3 so I used that to be consistent with the rest of the document.
